README.md

```diff
@@ -19,16 +19,47 @@ Once you have implemented the `fm` program (and tested it to the best of your ab
 
 ## Preprocessing
 
-*What preprocessing data do you store in files, and how?*
+The data that is stored is the processed information related to each genome passed from the '.fa' file. More specifically, the following information is stored:
+
+* rank_table: Table that indicates the rank for every letter of the alphabet at each point in the bwt string. Each letter counter gets increased downwards when the letter appears in bwt.
+* f vector: This is the ordered suffix array from the original string.
+* firstIndexList: It's a list that indicates the first time that each letter appears at the f column, in order to boost the indexation speed every iteration.
+* alphadic: A dictionary ordered alphabetically that stores the index of every letter.
 
 ## Insights you may have had while implementing the algorithm
 
+With the process of implementing the algorithm, we have realized the simplicity of its implementation, despite having to think a lot about it because it is very difficult to understand it properly. So, as a general conclusion, the implementation is very sotifiscated whilst elegant.
```

Comment:
That is indeed the case. It is tricky to understand but quite easy to implement.